packages/axes/src/types.ts

```diff
@@ -0,0 +1,122 @@
+import { SpringValues } from 'react-spring'
+import {
+    ScaleBand,
+    ScaleLinear,
+    ScaleOrdinal,
+    ScalePoint,
+    ScaleTime,
+    ScaleSymLog,
+    ScaleLogarithmic,
+} from 'd3-scale'
+import React from 'react'
+
+export type GridValuesBuilder<T> = T extends number
+    ? number[]
+    : T extends string
+    ? string[]
+    : T extends Date
+    ? Date[]
+    : never
+
+export type GridValues<T extends string | number | Date> = number | GridValuesBuilder<T>
+
+export type Point = {
+    x: number
+    y: number
+}
+
+export type ScaleWithBandwidth =
+    | (ScaleBand<any> & { type: 'band' })
+    | (ScalePoint<any> & { type: 'point' })
+
+export type AnyScale =
+    | (ScaleLinear<any, any> & { type: 'linear' })
```

Comment:
I think output is always a number, and input could be limited to `string | number | Date` as this is what's supported in this package.

Reply:
I've updated the output, but the `Domain` which is the first argument did not like the `string | number | Date`.